src/poly_utils/fold.rs

```diff
@@ -71,96 +67,97 @@ pub fn compute_fold<F: Field>(
-///
-/// 1. Transpose: reshape layout to enable independent processing of each sub-coset.
-/// 2. Inverse NTT: convert each sub-coset from evaluation to coefficient form (no 1/N scaling).
-/// 3. Scale correction:
-///    Each output is multiplied by:
-///
-///    ```ignore
-///    size_inv * (domain_gen_inv^i)^j
-///    ```
-///
-///    where:
-///      - `size_inv = 1 / 2^folding_factor`
-///      - `i` is the subdomain index
-///      - `j` is the index within the block
-///
-/// # Panics
-/// Panics if the input size is not divisible by `2^folding_factor`.
-#[cfg_attr(feature = "tracing", instrument(skip_all, fields(size = evals.len(), folding_factor)))]
-pub fn transform_evaluations<F: FftField>(
-    evals: &mut [F],
-    domain_gen_inv: F,
-    folding_factor: usize,
-) {
-    // Compute the number of sub-cosets = 2^folding_factor
-    let folding_factor_exp = 1 << folding_factor;
-
-    // Ensure input is divisible by folding factor
-    assert!(evals.len() % folding_factor_exp == 0);
-
-    // Number of rows (one per subdomain)
-    let size_of_new_domain = evals.len() / folding_factor_exp;
-
-    // Step 1: Reshape via transposition
-    transpose(evals, folding_factor_exp, size_of_new_domain);
-
-    // Step 2: Apply inverse NTTs
-    intt_batch(evals, folding_factor_exp);
-
-    // Step 3: Apply scaling to match the desired domain layout
-    // Each value is scaled by: size_inv * offset^j
-    let size_inv = F::from(folding_factor_exp as u64).inverse().unwrap();
-    #[cfg(not(feature = "parallel"))]
-    {
-        let mut coset_offset_inv = F::ONE;
-        for answers in evals.chunks_exact_mut(folding_factor_exp) {
-            let mut scale = size_inv;
-            for v in answers.iter_mut() {
-                *v *= scale;
-                scale *= coset_offset_inv;
-            }
-            coset_offset_inv *= domain_gen_inv;
-        }
-    }
-    #[cfg(feature = "parallel")]
-    evals
-        .par_chunks_exact_mut(folding_factor_exp)
-        .enumerate()
-        .for_each_with(F::ZERO, |offset, (i, answers)| {
-            if *offset == F::ZERO {
-                *offset = domain_gen_inv.pow([i as u64]);
-            } else {
-                *offset *= domain_gen_inv;
-            }
-            let mut scale = size_inv;
-            for v in answers.iter_mut() {
-                *v *= scale;
-                scale *= &*offset;
-            }
-        });
-}
-
 #[cfg(test)]
-mod tests {
+pub(crate) mod tests {
     use ark_ff::{AdditiveGroup, FftField, Field};
+    #[cfg(feature = "parallel")]
+    use rayon::prelude::*;
 
-    use super::{compute_fold, transform_evaluations};
+    use super::compute_fold;
     use crate::{
         crypto::fields::Field64,
-        ntt::transpose,
+        ntt::{intt_batch, transpose},
         poly_utils::{coeffs::CoefficientList, multilinear::MultilinearPoint},
     };
 
     type F = Field64;
 
+    /// Applies a folding transformation to evaluation vectors in-place.
```

Comment:
Let's add some note that this is only used to test consistency with new commit functionality